yt_dlp/extractor/scrolller.py

```diff
@@ -0,0 +1,101 @@
+from .common import InfoExtractor
+from ..utils import (determine_ext, KNOWN_EXTENSIONS)
+import json
+
+
+class ScrolllerIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?scrolller\.com/(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'https://scrolller.com/a-helping-hand-1k9pxikxkw',
+        'info_dict': {
+            'id': 'a-helping-hand-1k9pxikxkw',
+            'ext': "mp4",
+            'thumbnail': 'https://zepto.scrolller.com/a-helping-hand-3ty9q8x094-540x960.jpg',
+            'title': 'A helping hand',
+            "age_limit": 0,
+        }
+    }, {
+        'url': 'https://scrolller.com/tigers-chasing-a-drone-c5d1f2so6j',
+        'info_dict': {
+            "id": "tigers-chasing-a-drone-c5d1f2so6j",
+            'ext': "mp4",
+            'thumbnail': 'https://zepto.scrolller.com/tigers-chasing-a-drone-az9pkpguwe-540x303.jpg',
+            "title": "Tigers chasing a drone",
+            "age_limit": 0,
+        }
+    }, {
+        'url': 'https://scrolller.com/baby-rhino-smells-something-9chhugsv9p',
+        'info_dict': {
+            "id": "baby-rhino-smells-something-9chhugsv9p",
+            'ext': "mp4",
+            "thumbnail": 'https://atto.scrolller.com/hmm-whats-that-smell-bh54mf2c52-300x224.jpg',
+            "title": "Baby rhino smells something",
+            "age_limit": 0,
+        }
+    }, {
+        'url': 'https://scrolller.com/its-all-fun-and-games-cco8jjmoh7',
+        'info_dict': {
+            "id": "its-all-fun-and-games-cco8jjmoh7",
+            'ext': "mp4",
+            'thumbnail': 'https://atto.scrolller.com/its-all-fun-and-games-3amk9vg7m3-540x649.jpg',
+            "title": "It\'s all fun and games...",
+            "age_limit": 0,
+        }
+    }, {
+        'url': 'https://scrolller.com/may-the-force-be-with-you-octokuro-yeytg1fs7a',
+        'info_dict': {
+            "id": "may-the-force-be-with-you-octokuro-yeytg1fs7a",
+            'ext': "mp4",
+            'thumbnail': 'https://thumbs2.redgifs.com/DarkStarchyNautilus-poster.jpg',
+            "title": "May the force be with you (Octokuro)",
+            "age_limit": 18,
+        }
+    }
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        query = {
+            'query':
+            '''{
+                getSubredditPost(url:"/%s"){
+                    id
+                    title
+                    subredditTitle
+                    subredditUrl
+                    redditPath
+                    isNsfw
+                    mediaSources{
+                        url
+                        width
+                        height
```

Comment:
is it guaranteed these will always be int?

Reply:
I queried the graphQl api to get some data about the mutations and schema it uses and also get some data about the entry fields. The following fields indicate that width and height are scalar values, but I still added the int_or_none() guard to make sure there are no errors:

```json
{
      "name": "width",
      "args": [],
      "type": {
          "kind": "NON_NULL",
          "name": null,
          "ofType": {
              "kind": "SCALAR",
              "name": "Int",
              "ofType": null
          }
      },
      "isDeprecated": false,
      "deprecationReason": null
  },
  {
      "name": "height",
      "args": [],
      "type": {
          "kind": "NON_NULL",
          "name": null,
          "ofType": {
              "kind": "SCALAR",
              "name": "Int",
              "ofType": null
          }
      },
      "isDeprecated": false,
      "deprecationReason": null
  },
```